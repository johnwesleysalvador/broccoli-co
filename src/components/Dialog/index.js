import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import MaterialDialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Slide from '@material-ui/core/Slide';

function Transition(props) {
    return <Slide direction="down" {...props} />;
}

class Dialog extends PureComponent {
    state = {
        open: false,
    };

    componentWillReceiveProps({ open: nextOpen }) {
        if (nextOpen !== this.open) this.setState({ open: nextOpen });
    }

    handleClickOpen = () => {
        this.setState({ open: true });
    };

    handleClose = () => {
        const { onClose } = this.props;
        this.setState({ open: false });
        onClose();
    };

    render() {
        const { children, title } = this.props;
        return (
            <div>
                <MaterialDialog
                    open={this.state.open}
                    TransitionComponent={Transition}
                    keepMounted
                    onClose={this.handleClose}
                >
                    <DialogTitle id="alert-dialog-slide-title">{title}</DialogTitle>
                    <DialogContent>
                        <DialogContentText id="alert-dialog-slide-description">
                            {children}
                        </DialogContentText>
                    </DialogContent>
                    <DialogActions>
                        <Button onClick={this.handleClose} color="primary">
                            OK
                        </Button>
                    </DialogActions>
                </MaterialDialog>
            </div>
        );
    }
}

Dialog.propTypes = {
    open: PropTypes.bool,
    onClose: PropTypes.func,
    children: PropTypes.string,
    title: PropTypes.string,
};

Dialog.defaultProps = {
    open: false,
    onClose: () => {},
    title: '',
    children: '',
};

export default Dialog;
