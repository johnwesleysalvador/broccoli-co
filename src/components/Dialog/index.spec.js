import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import 'jest-styled-components';

import Dialog from '.';

describe('<Dialog />', () => {
    test('Should render Dialog with OPEN state', () => {
        const render = createShallow()(<Dialog open title="All done!" onClose={() => {}}>
                Lorem ipsum
        </Dialog>);
        expect(render).toMatchSnapshot();
    });
    test('Should render Dialog with CLOSE state', () => {
        const render = createShallow()(<Dialog open={false} title="All done!" onClose={() => {}}>
                Lorem ipsum
        </Dialog>);
        expect(render).toMatchSnapshot();
    });
});
