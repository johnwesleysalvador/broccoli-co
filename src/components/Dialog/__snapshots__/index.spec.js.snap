// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> Should render Dialog with CLOSE state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dialog
    onClose={[Function]}
    open={false}
    title="All done!"
>
    Lorem ipsum
</Dialog>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Dialog)
        TransitionComponent={[Function]}
        keepMounted={true}
        onClose={[Function]}
        open={false}
>
        <WithStyles(DialogTitle)
                id="alert-dialog-slide-title"
        >
                All done!
        </WithStyles(DialogTitle)>
        <WithStyles(DialogContent)>
                <WithStyles(DialogContentText)
                        id="alert-dialog-slide-description"
                >
                        Lorem ipsum
                </WithStyles(DialogContentText)>
        </WithStyles(DialogContent)>
        <WithStyles(DialogActions)>
                <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
                >
                        OK
                </WithStyles(Button)>
        </WithStyles(DialogActions)>
</WithStyles(Dialog)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "TransitionComponent": [Function],
        "children": Array [
          <WithStyles(DialogTitle)
            id="alert-dialog-slide-title"
>
            All done!
</WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)
                        id="alert-dialog-slide-description"
            >
                        Lorem ipsum
            </WithStyles(DialogContentText)>
</WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
            >
                        OK
            </WithStyles(Button)>
</WithStyles(DialogActions)>,
        ],
        "keepMounted": true,
        "onClose": [Function],
        "open": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "All done!",
            "id": "alert-dialog-slide-title",
          },
          "ref": null,
          "rendered": "All done!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(DialogContentText)
              id="alert-dialog-slide-description"
>
              Lorem ipsum
</WithStyles(DialogContentText)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Lorem ipsum",
              "id": "alert-dialog-slide-description",
            },
            "ref": null,
            "rendered": "Lorem ipsum",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              color="primary"
              onClick={[Function]}
>
              OK
</WithStyles(Button)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "OK",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "OK",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Dialog)
          TransitionComponent={[Function]}
          keepMounted={true}
          onClose={[Function]}
          open={false}
>
          <WithStyles(DialogTitle)
                    id="alert-dialog-slide-title"
          >
                    All done!
          </WithStyles(DialogTitle)>
          <WithStyles(DialogContent)>
                    <WithStyles(DialogContentText)
                              id="alert-dialog-slide-description"
                    >
                              Lorem ipsum
                    </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>
          <WithStyles(DialogActions)>
                    <WithStyles(Button)
                              color="primary"
                              onClick={[Function]}
                    >
                              OK
                    </WithStyles(Button)>
          </WithStyles(DialogActions)>
</WithStyles(Dialog)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "TransitionComponent": [Function],
          "children": Array [
            <WithStyles(DialogTitle)
              id="alert-dialog-slide-title"
>
              All done!
</WithStyles(DialogTitle)>,
            <WithStyles(DialogContent)>
              <WithStyles(DialogContentText)
                            id="alert-dialog-slide-description"
              >
                            Lorem ipsum
              </WithStyles(DialogContentText)>
</WithStyles(DialogContent)>,
            <WithStyles(DialogActions)>
              <WithStyles(Button)
                            color="primary"
                            onClick={[Function]}
              >
                            OK
              </WithStyles(Button)>
</WithStyles(DialogActions)>,
          ],
          "keepMounted": true,
          "onClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "All done!",
              "id": "alert-dialog-slide-title",
            },
            "ref": null,
            "rendered": "All done!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(DialogContentText)
                id="alert-dialog-slide-description"
>
                Lorem ipsum
</WithStyles(DialogContentText)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Lorem ipsum",
                "id": "alert-dialog-slide-description",
              },
              "ref": null,
              "rendered": "Lorem ipsum",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                color="primary"
                onClick={[Function]}
>
                OK
</WithStyles(Button)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "OK",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "OK",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;

exports[`<Dialog /> Should render Dialog with OPEN state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dialog
    onClose={[Function]}
    open={true}
    title="All done!"
>
    Lorem ipsum
</Dialog>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Dialog)
        TransitionComponent={[Function]}
        keepMounted={true}
        onClose={[Function]}
        open={false}
>
        <WithStyles(DialogTitle)
                id="alert-dialog-slide-title"
        >
                All done!
        </WithStyles(DialogTitle)>
        <WithStyles(DialogContent)>
                <WithStyles(DialogContentText)
                        id="alert-dialog-slide-description"
                >
                        Lorem ipsum
                </WithStyles(DialogContentText)>
        </WithStyles(DialogContent)>
        <WithStyles(DialogActions)>
                <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
                >
                        OK
                </WithStyles(Button)>
        </WithStyles(DialogActions)>
</WithStyles(Dialog)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "TransitionComponent": [Function],
        "children": Array [
          <WithStyles(DialogTitle)
            id="alert-dialog-slide-title"
>
            All done!
</WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)
                        id="alert-dialog-slide-description"
            >
                        Lorem ipsum
            </WithStyles(DialogContentText)>
</WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
            >
                        OK
            </WithStyles(Button)>
</WithStyles(DialogActions)>,
        ],
        "keepMounted": true,
        "onClose": [Function],
        "open": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "All done!",
            "id": "alert-dialog-slide-title",
          },
          "ref": null,
          "rendered": "All done!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(DialogContentText)
              id="alert-dialog-slide-description"
>
              Lorem ipsum
</WithStyles(DialogContentText)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Lorem ipsum",
              "id": "alert-dialog-slide-description",
            },
            "ref": null,
            "rendered": "Lorem ipsum",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              color="primary"
              onClick={[Function]}
>
              OK
</WithStyles(Button)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "OK",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "OK",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Dialog)
          TransitionComponent={[Function]}
          keepMounted={true}
          onClose={[Function]}
          open={false}
>
          <WithStyles(DialogTitle)
                    id="alert-dialog-slide-title"
          >
                    All done!
          </WithStyles(DialogTitle)>
          <WithStyles(DialogContent)>
                    <WithStyles(DialogContentText)
                              id="alert-dialog-slide-description"
                    >
                              Lorem ipsum
                    </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>
          <WithStyles(DialogActions)>
                    <WithStyles(Button)
                              color="primary"
                              onClick={[Function]}
                    >
                              OK
                    </WithStyles(Button)>
          </WithStyles(DialogActions)>
</WithStyles(Dialog)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "TransitionComponent": [Function],
          "children": Array [
            <WithStyles(DialogTitle)
              id="alert-dialog-slide-title"
>
              All done!
</WithStyles(DialogTitle)>,
            <WithStyles(DialogContent)>
              <WithStyles(DialogContentText)
                            id="alert-dialog-slide-description"
              >
                            Lorem ipsum
              </WithStyles(DialogContentText)>
</WithStyles(DialogContent)>,
            <WithStyles(DialogActions)>
              <WithStyles(Button)
                            color="primary"
                            onClick={[Function]}
              >
                            OK
              </WithStyles(Button)>
</WithStyles(DialogActions)>,
          ],
          "keepMounted": true,
          "onClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "All done!",
              "id": "alert-dialog-slide-title",
            },
            "ref": null,
            "rendered": "All done!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(DialogContentText)
                id="alert-dialog-slide-description"
>
                Lorem ipsum
</WithStyles(DialogContentText)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Lorem ipsum",
                "id": "alert-dialog-slide-description",
              },
              "ref": null,
              "rendered": "Lorem ipsum",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                color="primary"
                onClick={[Function]}
>
                OK
</WithStyles(Button)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "OK",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "OK",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
